WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -3049,6 +3065,51 @@ public void onTrackableEvent(TrackableEvent event, Map<String, String> propertie
         mEditorTracker.trackEditorEvent(event, mEditorFragment.getEditorName(), properties);
     }
 
+    @Override public void onStoryComposerLoadRequested(ArrayList<Object> mediaFiles, String blockId) {
```

Comment:
Hi @mzorz I can see where this PR and the follow up is adding more code to this activity. Let's try to refactor some of this logic out into another class so we don't make this class any bigger since it's in the 3000s now 😅 All the `dispatcher`/ `EventBus` related logic can be placed in another class like how we did it [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/PostListEventListener.kt)
If you have a reason why it has to be placed here then we can leave it but if we can extract then that would be good. Wdyt? 😄

Reply:
That's a smart idea Joel! (using the lifecycleobserver and externalizing the logic for eventbus register/unregister and listening out to a different class, as shown in the PostsListEvent example). In this case, `onStoryComposerLoadRequested` is not part of EventBus / dispatcher code, it's an interface implementation which connects Gutenberg to the containing Activity (EditPostActivity). So, the interface override I'd say needs to stay there, but we could move the logic within that method override out, to keep EditPostActivity shorter.

Done in 0f4d87c25b 👍

Reply:
forgot to mention -  looking into the other PR, I'm creating the suggested class and refactor the listeners there in #12982 👍

Reply:
the suggested refactor effectively made available in the PR after this one (#12982) in this commit https://github.com/wordpress-mobile/WordPress-Android/pull/12982/commits/a6dee817a13eddd0aecfbda821b5d0767dc8f508 @jd-alexander 👍

Reply:
Nice work @mzorz. Thanks for this change 🙇